### ISSUE TITLE
accessNotConfigured: Project 294899064488 has been scheduled for deletion

### ISSUE THREAD

**Issue by viollinn** [enen92/script.tubecast#38]
Hello, when i try to use this addon on Kodi, i've got the following message:
``
2020-02-29 21:57:53.240 T:1436541824   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.Exception'>
                                            Error Contents: Error reason: |accessNotConfigured| with message: |Project 294899064488 has been scheduled for deletion and cannot be used for API calls. Visit https://console.developers.google.com/iam-admin/projects?pendingDeletion=true to undelete the project.|
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/default.py", line 15, in <module>
                                                runner.run(__provider__)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/runner.py", line 60, in run
                                                __RUNNER__.run(provider, context)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/impl/xbmc/xbmc_runner.py", line 33, in run
                                                results = provider.navigate(context)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/abstract_provider.py", line 99, in navigate
                                                result = method(context, re_match)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/kodion/register_provider_path.py", line 19, in wrapper
                                                return func(*args, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/provider.py", line 749, in on_play
                                                return yt_play.play_video(self, context)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/yt_play.py", line 86, in play_video
                                                use_play_data=playback_history)
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/utils.py", line 411, in update_play_info
                                                video_data = resource_manager.get_videos([video_id])
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/resource_manager.py", line 138, in get_videos
                                                result.update(self._update_videos(list_of_50, live_details))
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/resource_manager.py", line 121, in _update_videos
                                                if self.handle_error(json_data):
                                              File "/storage/.kodi/addons/plugin.video.youtube/resources/lib/youtube_plugin/youtube/helper/resource_manager.py", line 240, in handle_error
                                                raise Exception(error_message)
                                            Exception: Error reason: |accessNotConfigured| with message: |Project 294899064488 has been scheduled for deletion and cannot be used for API calls. Visit https://console.developers.google.com/iam-admin/projects?pendingDeletion=true to undelete the project.|
                                            -->End of Python script error report<--
2020-02-29 22:05:48.666 T:1937652160   ERROR: Control 1000 in window 13001 has been asked to focus, but it can't
``

Can u do something for that ? Related to Google API ?

Thnx

**Comment by siku2** [comment]
@viollinn this is an issue with the underlying YouTube plugin. It's already being tracked [here](https://github.com/jdf76/plugin.video.youtube/issues/736) (https://github.com/jdf76/plugin.video.youtube/issues/736).
TL;DR you have to set up your own API keys to fix this.

**Comment by i3laze** [comment]
Seems, both API Key and Client OAuth ID+Secret are required.
How-to for the first one here:
https://www.slickremix.com/docs/get-api-key-for-youtube/

**Comment by enen92** [comment]
Not really, if the youtube addon doesn't have any credentials configured it will still allow playback requests from tubecast. You'll loose access to the data API, which means the played videos won't have any thumbs, title, etc.
